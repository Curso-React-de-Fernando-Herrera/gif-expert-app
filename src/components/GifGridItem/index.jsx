import PropTypes from 'prop-types'

const GifGridItem = ({ title, url, username }) => {
  return <article>
    <h1>{title}</h1>
    <small>{username}</small>
    <img src={url} alt={title} loading="lazy" />
  </article>
}

GifGridItem.propTypes = {
  title: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired,
  username: PropTypes.string.isRequired
}

export default GifGridItem
