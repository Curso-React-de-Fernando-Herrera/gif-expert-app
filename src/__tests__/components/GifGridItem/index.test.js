import { shallow } from 'enzyme'
import GifGridItem from 'components/GifGridItem'

describe('<GifGridItem /> component', () => {
  const title = 'Gif Title'
  const username = 'GifyUserCrazy'
  const url = 'https://example'
  const wrapper = shallow(
    <GifGridItem
      title={title}
      username={username}
      url={url}
    />
  )

  test('should be rendered', () => {
    expect(wrapper).toMatchSnapshot()
  })

  test('should be same title that arrive by prop', () => {
    const returnedTitle = wrapper
      .find('h1')
      .text()
      .trim()
    
    expect(returnedTitle).toBe(title)
  })

  test('should be same imgs atributes that arrive by props', () => {
    const returnedImg = wrapper
      .find('img')
      .props()
      .src
    
    expect(returnedImg).toBe(url)
  })
})