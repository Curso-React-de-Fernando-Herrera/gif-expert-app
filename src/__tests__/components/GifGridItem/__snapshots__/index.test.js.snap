// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GifGridItem /> component should be rendered 1`] = `
<article>
  <h1>
    Gif Title
  </h1>
  <small>
    GifyUserCrazy
  </small>
  <img
    alt="Gif Title"
    loading="lazy"
    src="https://example"
  />
</article>
`;
